inference-engine/include/cpp/ie_cnn_network.h

```diff
@@ -253,6 +262,10 @@ class INFERENCE_ENGINE_API_CLASS(CNNNetwork) {
      * @brief A pointer to output data
      */
     DataPtr output;
+    /**
+     * @brief A pointer to weights data
+    */
+    std::shared_ptr<IAllocator> weights;
```

Comment:
Is it correct place for weights allocator? The ICNNNetwork and weights lifetime should be the same. If any plugin saves the ICNNNetwork pointer and at the same time, CNNNetwork is destroyed, we may crash.